### PR TITLE
Test against older python and C# versions

### PR DESCRIPTION
**Pull request by chriselion**
### Proposed change(s)

Extends yamato tests to use older versions of python and C# codebase. 

### Types of change(s)

- [ ] Bug fix
- [ ] New feature
- [ ] Code refactor
- [ ] Breaking change
- [ ] Documentation update
- [ ] Other (please describe)

### Checklist
- [ ] Added tests that prove my fix is effective or that my feature works
- [ ] Updated the [changelog](https://github.com/Unity-Technologies/ml-agents/blob/master/com.unity.ml-agents/CHANGELOG.md) (if applicable)
- [ ] Updated the [documentation](https://github.com/Unity-Technologies/ml-agents/tree/master/docs) (if applicable)
- [ ] Updated the [migration guide](https://github.com/Unity-Technologies/ml-agents/blob/master/docs/Migrating.md) (if applicable)

### CODE REVIEW

**Comment by chriselion** [comment]
Confirmed that 0.14.0 versions of C# and python failed (although not quite the way I would have expected):
https://github.com/Unity-Technologies/ml-agents/pull/3693
https://github.com/Unity-Technologies/ml-agents/pull/3694

**Comment by chriselion** [comment]
@vincentpierre followup from the meeting - 0.14 used the old format for the communication protocol versioning, so there's no expectation that it'll work with master (or 0.15):
https://github.com/Unity-Technologies/ml-agents/blob/0.14.0/ml-agents-envs/mlagents_envs/environment.py#L54

File: ml-agents/tests/yamato/yamato_utils.py [hunk 1]
```diff
@@ -49,21 +49,62 @@ def run_standalone_build(base_path: str, verbose: bool = False) -> int:
...
+        subprocess.check_call(
+            f"git checkout {csharp_version} -- {csharp_dir}", shell=True
+        )
+
+
+def undo_git_checkout():
+    """
+    Clean up the git working directory.
+    """
+    subprocess.check_call("git reset HEAD .", shell=True)
```
Comment (chriselion) [inline ml-agents/tests/yamato/yamato_utils.py]: Number of times I accidentally reverted changes to this file while testing these commands: 2

Comment (surfnerd) [reply ml-agents/tests/yamato/yamato_utils.py]: I created a git utility class while working at nvidia in order to be able to mock out the git calls for testing.  saved my life.

Comment (chriselion) [reply ml-agents/tests/yamato/yamato_utils.py]: YAGNI for now, but it if we need to do more git logic, I'll look into it.

**Pull request merged**